src/App.tsx

```diff
@@ -15,48 +17,143 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET: {
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    }
+
+    case SortType.LENGTH: {
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    }
+
+    case SortType.NONE:
+    default: {
+      break;
+    }
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
```

Comment:
```suggestion
  const sortBy = (sortType: SortType) => () => {
    setSortType(sortType);
  };
```
functions with same logic. 
it can be one handler which will be accept sortType and return arrow function.
and just call in JSX